ark/utils/load_utils_test.py

```diff
@@ -206,60 +186,72 @@ def test_load_imgs_from_dir():
+                                          trim_suffix='_')
         with pytest.raises(ValueError):
-            load_utils.load_imgs_from_dir(temp_dir, files=['not_an_image'],
-                                          trim_suffix='_', dtype=np.float32)
+            load_utils.load_imgs_from_dir(temp_dir, files=['not_an_image'], trim_suffix='_')
 
         # check default loading
         loaded_xr = load_utils.load_imgs_from_dir(temp_dir, trim_suffix='_',
-                                                  xr_dim_name='compartments', dtype=np.float32)
+                                                  xr_dim_name='compartments')
 
         assert loaded_xr.equals(data_xr)
 
         # check suffix matched loading:
         loaded_xr = load_utils.load_imgs_from_dir(temp_dir, match_substring='_otherinfo',
-                                                  trim_suffix='_', xr_dim_name='compartments',
-                                                  dtype=np.float32)
-
+                                                  trim_suffix='_', xr_dim_name='compartments')
         assert loaded_xr.equals(data_xr.loc[['fov0'], :, :, :])
 
         fovnames = [f'{fov}.tiff' for fov in fovs]
 
         # check general substring matched loading
         loaded_xr = load_utils.load_imgs_from_dir(temp_dir, match_substring='ov', trim_suffix='_',
-                                                  xr_dim_name='compartments', dtype=np.float32)
+                                                  xr_dim_name='compartments')
 
         assert loaded_xr.equals(data_xr)
 
         # check provided file overruling of match_substring
         loaded_xr = load_utils.load_imgs_from_dir(temp_dir, files=fovnames,
                                                   match_substring='_otherinfo', trim_suffix='_',
-                                                  xr_dim_name='compartments', dtype=np.float32)
+                                                  xr_dim_name='compartments')
 
         assert loaded_xr.equals(data_xr)
 
         # test error on no matched suffix
         with pytest.raises(ValueError):
             load_utils.load_imgs_from_dir(temp_dir, match_substring='not_a_real_suffix',
-                                          trim_suffix='_', xr_dim_name='compartments',
-                                          dtype=np.float32)
+                                          trim_suffix='_', xr_dim_name='compartments')
+
+    # Test floating point xarray type consistency (i.e. creation type == load_imgs type)
+    with tempfile.TemporaryDirectory() as temp_dir:
+
+        fovs, _ = test_utils.gen_fov_chan_names(num_fovs=3, num_chans=0, use_delimiter=True)
+        filelocs, data_xr = test_utils.create_paired_xarray_fovs(temp_dir, fovs, [0],
+                                                                 img_shape=(10, 10), mode='labels',
+                                                                 delimiter='_', dtype=np.float32)
```

Comment:
This and the one below can get turned into a for loop over dtype creation values